### PR TITLE
[14.0][ADD] auth_device - Allows to connect with an external device whose the value is stored on the user

### PR DESCRIPTION
**Pull request by FrancoMaxime**
Hi,
This module adds a functionality allowing to connect with an external device (like an RFID card, a badge,...). It assumes there is a mechanism able to read this external device and that is connected to the computer so it can insert the value of the external device inside the connection form.

I ask myself the question: "should I add some JavaScript code to send directly the form after the user input instead of pressing the button login" ?

All your feedback and/or code review are welcome

### CODE REVIEW

**Review by apineux — APPROVED** [review]

File: auth_device/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Auth Device",
+    "summary": "Allow users to login through an externel device.",
```
Comment (baimont) [inline auth_device/__manifest__.py]:
```suggestion
    "summary": "Allows users to log in through an external device.",
```

File: auth_device/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+from odoo import _, http
+from odoo.exceptions import AccessDenied
+from odoo.http import request
+
+from odoo.addons.portal.controllers.web import Home
+from odoo.addons.web.controllers.main import ensure_db
+
+
+class DeviceController(Home):
```
Comment (baimont) [inline auth_device/controllers/main.py:15]: pour codecov il faudrait des tests unitaires sur le contrôleur

**Comment by baimont** [comment]
LGTM!

**Comment by rousseldenis** [comment]
@FrancoMaxime Could you attend comments ?

**Pull request closed without merging**